src/sentry/web/frontend/oauth_token.py

```diff
@@ -42,28 +45,46 @@ def error(self, request: HttpRequest, name, reason=None, status=400):
 
     @never_cache
     def post(self, request: HttpRequest) -> HttpResponse:
+        client_details = self._get_client_details_if_valid_request(request=request)
+
+        if not isinstance(client_details, dict):
+            return client_details  # client_details contains an error response.
+
+        if client_details["grant_type"] == GrantTypes.AUTHORIZATION:
+            return self._get_access_tokens(
+                request=request, application=client_details["application"]
+            )
+        return self._get_refresh_token(request, application=client_details["application"])
+
+    def _get_client_details_if_valid_request(self, request: Request) -> Union[dict, HttpResponse]:
+        """Returns the client's ID and associated ApiApplication if they provided the correct client secret."""
         grant_type = request.POST.get("grant_type")
         client_id = request.POST.get("client_id")
         if not client_id:
             return self.error(request, "invalid_client", "missing client_id")
-        if grant_type == GrantTypes.AUTHORIZATION:
-            return self._get_access_tokens(request, client_id)
-        elif grant_type == GrantTypes.REFRESH:
-            return self._get_refresh_token(request, client_id)
-        else:
+        if grant_type not in [GrantTypes.AUTHORIZATION, GrantTypes.REFRESH]:
             return self.error(request, "unsupported_grant_type")
-
-    def _get_access_tokens(self, request, client_id):
-        redirect_uri = request.POST.get("redirect_uri")
-        code = request.POST.get("code")
-
         try:
             application = ApiApplication.objects.get(
                 client_id=client_id, status=ApiApplicationStatus.active
             )
         except ApiApplication.DoesNotExist:
             return self.error(request, "invalid_client", "invalid client_id")
 
+        # date that we started requiring client secrets
+        cutoff_date = datetime.datetime(2023, 7, 16).astimezone()
```

Comment:
If we're hardcoding a date we might as well use utc and epoch 🤷

Reply:
could you suggest a change or point to somewhere in the codebase where it's done like this? Not sure what you're looking for

Reply:
Oh - If you do `datetime(asdf).time()` it will give you a timestamp that's in epoch - which is the number of milliseconds passed since Jan 1 1970 UTC 😂 

For this application probably doens't matter since we'll be past the date anyways - but epoch makes it easier when you don't want to deal with timezones and such

Reply:
you can also do `datetime(2023, 7, 16, 0, 0, 0, tzinfo=timezone.utc)`